doc/release-notes.md

```diff
@@ -1,97 +1,384 @@
-Dash Core version 0.12.1 is now available from:
+Dash Core version 0.12.2
+========================
 
-  <https://www.dash.org/downloads/>
+Release is now available from:
 
+  <https://www.dash.org/downloads/#wallets>
 
+This is a new major version release, bringing new features and other improvements.
 
+Please report bugs using the issue tracker at github:
 
-Older releases
---------------
+  <https://github.com/dashpay/dash/issues>
 
-Dash was previously known as Darkcoin.
+Upgrading and downgrading
+=========================
 
-Darkcoin tree 0.8.x was a fork of Litecoin tree 0.8, original name was XCoin
-which was first released on Jan/18/2014.
+How to Upgrade
+--------------
 
-### Downgrade to a version < 0.12.0
+If you are running an older version, shut it down. Wait until it has completely
+shut down (which might take a few minutes for older versions), then run the
+installer (on Windows) or just copy over /Applications/Dash-Qt (on Mac) or
+dashd/dash-qt (on Linux).
 
-Because release 0.12.0 and later will obfuscate the chainstate on every
-fresh sync or reindex, the chainstate is not backwards-compatible with
-pre-0.12 versions of Bitcoin Core or other software.
+Downgrade warning
+-----------------
+### Downgrade to a version < 0.12.2
 
-If you want to downgrade after you have done a reindex with 0.12.0 or later,
-you will need to reindex when you first start Bitcoin Core version 0.11 or
-earlier.
+Because release 0.12.2 includes DIP0001 (2 MB block size hardfork) plus
+a transaction fee reduction and a fix for masternode rank calculation algo
+(which activation also depends on DIP0001) this release will not be
+backwards compatible after DIP0001 lock in/activation happens.
+
+This does not affect wallet forward or backward compatibility.
 
 Notable changes
 ===============
 
-Example item
----------------------------------------
+DIP0001
+-------
 
-Example text.
+We outline an initial scaling mechanism for Dash. After deployment and activation, Dash will be able to handle double the transactions it can currently handle. This means that Dash will be prepared for eight times the traffic of Bitcoin.
```

Comment:
It may be unclear to some people why a doubling in block size equals 8x TX throughput. Maybe something like this instead of the last sentence:
Together with the faster block times, Dash we will be prepared to handle eight times the traffic of Bitcoin.

Reply:
Good point. And also this comparison to Bitcoin doesn't account for segwit txes. Though with current 10% adoption it makes not that much difference yet, so probably we are not that off.